test/test_dists.jl

```diff
@@ -1051,6 +1051,124 @@ end
+            @test_throws DimensionMismatch dist(x, x)
+            @test_throws BoundsError D(x, x)  # Could choose to special case this?
+        elseif dist isa Union{BhattacharyyaDist, HellingerDist, MeanAbsDeviation, MeanSqDeviation, RMSDeviation}
+            @test isnan(dist(x, x))
+            @test isnan(D(x, x))
+        else
+            @test D(x, x) == dist(x, x)
+        end
+
+        # Cover existing failure cases with missings
+        # TODO: Simplify this with an error variable
+        if dist isa Union{Hamming, WeightedHamming, ChiSqDist, KLDivergence, GenKLDivergence, SpanNormDist}
+            @test_throws TypeError dist(a, b)
+            @test_throws TypeError colwise(dist, A, B)
+            @test_throws TypeError pairwise(dist, A, B, dims=2)
+            @test_throws TypeError pairwise(dist, A, dims=2)
+        elseif dist isa JSDivergence
+            @test_throws TypeError dist(a, b)
+            @test_throws MethodError colwise(dist, A, B)
+            @test_throws MethodError pairwise(dist, A, B, dims=2)
+            @test_throws MethodError pairwise(dist, A, dims=2)
+        elseif dist isa Bregman
+            @test_throws ArgumentError dist(a, b)
+            @test_throws ArgumentError colwise(dist, A, B)
+            @test_throws ArgumentError pairwise(dist, A, B, dims=2)
+            @test_throws ArgumentError pairwise(dist, A, dims=2)
+        elseif dist isa CorrDist
+            @test_throws UndefVarError dist(a, b)
+            @test_throws UndefVarError colwise(dist, A, B)
+            @test_throws MethodError pairwise(dist, A, B, dims=2)
+            @test_throws MethodError pairwise(dist, A, dims=2)
+        elseif dist isa RenyiDivergence
+            @test_throws MethodError dist(a, b)
+            @test_throws MethodError colwise(dist, A, B)
+            @test_throws MethodError pairwise(dist, A, B, dims=2)
+            @test_throws MethodError pairwise(dist, A, dims=2)
+
+            # Doesn't handle eltype Union{T, Missing}
+            @test_throws MethodError colwise(dist, A[:, 3:4], B[:, 3:4])
+        else
+            @test ismissing(dist(a, b))
+            @test_throws MethodError colwise(dist, A, B)
+            @test_throws MethodError pairwise(dist, A, B, dims=2)
+            @test_throws MethodError pairwise(dist, A, dims=2)
+        end
+
+        # Handle weights
+        if dist isa Distances.UnionMetrics && Distances.parameters(dist) isa Vector
+            @test D(a, b) == Distances._evaluate(dist, [1, 5], [6, 10], Distances.parameters(dist)[[1, 5]])
+        else
+            @test D(a, b) == dist([1, 5], [6, 10])
+        end
+
+        @test colwise(D, A, B)[3:4] ≈ colwise(dist, disallowmissing(A[:, 3:4]), disallowmissing(B[:, 3:4]))
+
+        M = pairwise(D, A, B, dims=2)
+        @test size(M) == (4, 4)
+        @test !any(ismissing, M)
+
+        M = pairwise(D, A, dims=2)
+        @test size(M) == (4, 4)
+        @test !any(ismissing, M)
```

Comment:
Could manually check the calculations here, but I figured a few smoke tests that it doesn't error or return `missing`s was good enough?